src/service/extension-location.js

```diff
@@ -105,10 +115,37 @@ export function calculateEntryPointScriptUrl(
 export function parseExtensionUrl(scriptUrl) {
   // Note that the "(\.max)?" group only applies to local dev.
   const matches = scriptUrl.match(
-    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)$/i
+    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)(\?.*)?$/i
   );
   return {
     extensionId: matches ? matches[2] : undefined,
     extensionVersion: matches ? matches[3] : undefined,
   };
 }
+
+/**
+ * Based on mode, determine which file extension to use for the script URL.
+ * @param {boolean=} opt_isLocalDev
+ * @return {string}
+ */
+function calculateFileExtension(opt_isLocalDev) {
+  if (getMode().sxg && opt_isLocalDev) {
+    return '.sxg.js';
+  } else if (getMode().sxg || getMode().esm) {
+    return '.mjs';
```

Comment:
Shouldn't we be using  `.js` for SXG builds?

Reply:
Refer to #30126. We're outputting `.sxg.js` in `dist` and `release`.

Reply:
Yes, but here we're checking for `getMode().sxg` and inserting a `.mjs` file extension. I think this should be injecting `.js` if we're in sxg mode.